src/python/pants/backend/python/rules/python_create_binary.py

```diff
@@ -2,28 +2,25 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 from dataclasses import dataclass
-from typing import Optional
+from typing import ClassVar, Optional
 
 from pants.backend.python.rules.pex import Pex
 from pants.backend.python.rules.pex_from_target_closure import CreatePexFromTargetClosure
 from pants.backend.python.rules.targets import EntryPoint, PythonBinarySources
 from pants.backend.python.targets.python_binary import PythonBinary
 from pants.build_graph.address import Address
 from pants.engine.addressable import Addresses
-from pants.engine.legacy.structs import PythonBinaryAdaptor
 from pants.engine.rules import UnionRule, rule
 from pants.engine.selectors import Get
-from pants.engine.target import Target, WrappedTarget
-from pants.rules.core.binary import BinaryTarget, CreatedBinary
+from pants.engine.target import Target
+from pants.rules.core.binary import BinaryImplementation, CreatedBinary
 from pants.rules.core.strip_source_roots import SourceRootStrippedSources, StripSourcesFieldRequest
 
 
-# TODO: consider replacing this with sugar like `SelectFields(EntryPoint, PythonBinarySources)` so
-#  that the rule would request that instead of this dataclass. Note that this syntax must support
-#  both optional_fields (see the below TODO) and opt-out `SentinelField`s
-#  (see https://github.com/pantsbuild/pants/pull/9316#issuecomment-600152573).
```

Comment:
This TODO is superseded by a new TODO in `binary.py`. The abstraction has now moved one layer up.